jf_agent/main.py

```diff
@@ -567,19 +552,40 @@ def _s3_cmd(cmd):
 
     print('Sending data to Jellyfish... ')
 
-    output_basedir, output_dir_timestamp = os.path.split(outdir)
-    s3_uri_prefix_with_timestamp = os.path.join(s3_uri_prefix, output_dir_timestamp)
-    done_file_path = f'{os.path.join(outdir, ".done")}'
+    signed_urls = get_signed_url(os.listdir(config.outdir))
+
+    threads = [
+        threading.Thread(
+            target=upload_file,
+            args=[index, filename, file_dict['s3_path'], file_dict['url']],
+        )
+        for index, (filename, file_dict) in enumerate(signed_urls.items())
+    ]
+
+    thread_exceptions = [None] * (len(signed_urls))
+    
+    print(f'Starting {len(threads)} threads')
+
+    for t in thread:
+        t.start()
+
+    for t in threads:
+        t.join()
+
+    if any(thread_exceptions):
+        print(f'ERROR: not all files uploaded to S3. Files have been saved locally. Once connectivity issues are resolved, try running the Agent in send_only mode.')
+        return
+
+    # creating .done file
+    done_file_path = f'{os.path.join(config.outdir, ".done")}'
     if os.path.exists(done_file_path):
         print(
             f'ERROR: {done_file_path} already exists -- has this data already been sent to Jellyfish?'
```

Comment:
I'm not sure this check is helpful.

In the case that the S3 upload of the .done file fails, you're left with a .done file sitting on the disk with the other generated files; if you try to send again, we'd hit this error, which seems unhelpful.  Why not just let them re-send?